vevestaX/vevesta.py

```diff
@@ -780,13 +803,34 @@ def __getExcelSheetData(self, fileName, sheetName):
                 modelingData = pandas.read_excel(fileName, sheet_name=sheetName, index_col=[])
                 return modelingData
 
-    def commit(self, techniqueUsed, filename=None, message=None, version=None, projectId=None, attachmentFlag=True):
+    def commit(self, techniqueUsed, filename=None, message=None, version=None, projectId=None, attachmentFlag=True,
+               repoName=None, sourceBranch=None, newBranch=None):
         self.dump(techniqueUsed, filename=filename, message=message, version=version, showMessage=False)
 
         # api-endpoint
-        token = open("access_token.txt", "r").read()
+        token = self.__find_access_token()
         backend_url = 'https://api.matrixkanban.com/services-1.0-SNAPSHOT'
 
+        headers = {'Authorization': 'Bearer ' + token}
+        params = {'projectId': projectId}
+
+        # get git access token
+        response = requests.post(url=backend_url + '/GetGitToken', headers=headers, params=params)
+        data = response.json()
+        git_token = data['gitToken']
+        project_title = data['projectTitle']
+        if repoName is None:
+            repoName = project_title
+        if sourceBranch is None:
```

Comment:
this isnt correct... source branch is master branch on which nothing will be pushed except when it has been specified also by the user as newbranch